### PR TITLE
Intent to implement Ad-Type Adverticum.

### PR DESCRIPTION
**Pull request by Adverticum**
The implementation of AdverticumAdServer to the AMP Supported Networks.

### CODE REVIEW

**Comment by Adverticum** [comment]
Thank you for your contribution and review. Do you know when will our integration go live? Do we have anything else to do with the project or it's all on you now?

Thanks again,
Teleki Sándor

File: ads/_config.js [hunk 1]
```diff
@@ -254,10 +256,7 @@ export const adConfig = {
     renderStartImplemented: true,
   },
 
-  ligatus: {
-    prefetch: 'https://ssl.ligatus.com/render/ligrend.js',
-    renderStartImplemented: true,
-  },
+  ligatus: {},
```
Comment (lannka) [inline ads/_config.js]: please keep this

File: ads/adverticum.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  if (data.goa3zone) {
+    const zoneid = 'zone' + data.goa3zone;
+    const d = global.document.createElement('div');
+
+
+    d.setAttribute('id', zoneid);
+    d.setAttribute('class', 'goAdverticum goa-embedded');
+
+
+
```
Comment (lannka) [inline ads/adverticum.js]: remove extra blank lines

File: examples/ads.amp.html [hunk 3]
```diff
@@ -161,9 +162,10 @@
     <a href="#yieldmo">Yieldmo</a> |
     <a href="#zedo">ZEDO</a> |
     <a href="#zergnet">ZergNet</a> |
-    <a href="#zucks">Zucks</a> |
+    <a href="#zucks">Zucks</a> | -->
```
Comment (lannka) [inline examples/ads.amp.html]: revert this

File: 3p/integration.js [hunk 4]
```diff
@@ -125,6 +126,8 @@ import {zedo} from '../ads/zedo';
 import {zergnet} from '../ads/zergnet';
 import {zucks} from '../ads/zucks';
 
+
+
```
Comment (zhouyx) [inline 3p/integration.js]: please revert this 2 empty line change

File: 3p/integration.js [hunk 5]
```diff
@@ -149,6 +152,7 @@ if (getMode().test || getMode().localDev) {
 }
 
 // Keep the list in alphabetic order
+register('adverticum',adverticum);
```
Comment (zhouyx) [inline 3p/integration.js]: alphabetical order please

File: ads/_config.js [hunk 6]
```diff
@@ -40,6 +40,7 @@ export const adConfig = {
     renderStartImplemented: true,
   },
 
+
```
Comment (zhouyx) [inline ads/_config.js]: unnecessary change.

File: ads/_config.js [hunk 7]
```diff
@@ -110,6 +111,7 @@ export const adConfig = {
     ],
     renderStartImplemented: true,
   },
+  adverticum: {},
```
Comment (zhouyx) [inline ads/_config.js]: please add empty line above

File: ads/adverticum.js [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+import {writeScript} from '../3p/3p';
+
+
+/**
+ * @param {!Window} global
+ * @param {!Object} data
+ */
+
+export function adverticum(global,data) {
```
Comment (zhouyx) [inline ads/adverticum.js]: nit: space between `global` and `data`

File: ads/adverticum.md [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+
+
+##Example
+
+
+<amp-ad width=300 height=250
+    type="adverticum" data-goa3zone="123456789" data-costumetargetstring="bXVzdGJlYmVhc2U2NGVuY29kZWQ=" >
+</amp-ad>
+
+Note: The only supported costume parameters are:
```
Comment (zhouyx) [inline ads/adverticum.md]: You can add a configuration part, and list supported parameters.

File: ads/adverticum.md [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+-->
+
+#Adverticum
+
+
+
+##Example
+
+
+<amp-ad width=300 height=250
```
Comment (zhouyx) [inline ads/adverticum.md]: I can't see the example. Please follow correct style. Here's one example.
https://raw.githubusercontent.com/ampproject/amphtml/master/ads/google/adsense.md

File: examples/ads.amp.html [hunk 11]
```diff
@@ -161,9 +162,10 @@
...
   </nav>
 
+
```
Comment (zhouyx) [inline examples/ads.amp.html]: please revert empty line

File: ads/adverticum.md (hunk 10 above)
Comment (zhouyx) [inline ads/adverticum.md]: I can't see the example in preview. Please follow correct style. 
One example https://raw.githubusercontent.com/ampproject/amphtml/master/ads/google/adsense.md

File: ads/adverticum.md (hunk 9 above)
Comment (zhouyx) [inline ads/adverticum.md]: I would recommend add a configuration part, and list all supported parameters there.

File: ads/adverticum.js [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+  if (data.goa3zone) {
```
Comment (zhouyx) [inline ads/adverticum.js]: You can use `#validateData` function to do this.

File: ads/adverticum.md [hunk 13]
```diff
@@ -0,0 +1,37 @@
+
```
Comment (zhouyx) [inline ads/adverticum.md]: remove this empty line

**Comment by zhouyx** [comment]
@Adverticum Thanks for the Pull Request. Some requests before merge

**Comment by Adverticum** [comment]
Hy guys!

Can we get a review please ?:)

File: ads/adverticum.js [hunk 14]
```diff
@@ -0,0 +1,46 @@
...
+    d.setAttribute('id', zoneid);
+    d.setAttribute('class', 'goAdverticum goa-embedded');
+
+    document.getElementById('c').appendChild(d);
+
+    if (data.costumetargetstring) {
+      const s = global.document.createTextNode(data.costumetargetstring);
+      const v = global.document.createElement('var');
+
+      v.setAttribute('id','cT');
```
Comment (zhouyx) [inline ads/adverticum.js]: nit: space after ,

File: examples/ads.amp.html [hunk 15]
```diff
@@ -317,6 +318,14 @@ <h2 id="adverline">Adverline</h2>
     <div fallback></div>
   </amp-ad>
 
+  <amp-ad width=300 height=250
+     type="adverticum"
```
Comment (zhouyx) [inline examples/ads.amp.html]: I am not able to see a rendered ad example. Got Error: `timeout wait for ad response`. Please check and update the example.

**Comment by Adverticum** [comment]
I started to feel a bit flooded and messed up things. Started a new PR:
https://github.com/ampproject/amphtml/pull/6493

Please continue there.
